AmplifyPlugins/Storage/AWSS3StoragePlugin/Operation/AWSS3StorageUploadFileOperation.swift

```diff
@@ -127,6 +129,18 @@ public class AWSS3StorageUploadFileOperation: AmplifyInProcessReportingOperation
         }
     }
 
+    private func resolvePrefix(accessLevel: StorageAccessLevel,
+                               targetIdentityId: String?,
+                               completion: @escaping (Result<String, StorageError>) -> Void) {
+        let prefixResolver = storageConfiguration.prefixResolver ??
+        StorageAccessLevelAwarePrefixResolver(authService: authService)
+        DispatchQueue.global().async {
```

Comment:
can we run this on different queue? We should avoid using `DispatchQueue.global()`